mirrord-layer/src/pod_api.rs

```diff
@@ -162,6 +163,14 @@ pub async fn create_agent(
 
     let _ = tokio::time::timeout(std::time::Duration::from_secs(20), running)
         .await
-        .unwrap();
-    pods_api.portforward(&pod_name, &[61337]).await.unwrap()
+        .with_context(|| {
+            format!(
+                "Failed to receive a timely response from pod: {:?}",
+                pod_name
+            )
+        });
```

Comment:
you lack `?` at the end of the line to make it return the error.

Reply:
Your wish is my command